htdocs/plugins.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * OpenWebRx+ Plugin loader
+ *
+ * You should load your plugins in "plugins/{type}/init.js",
+ * where {type} is 'receiver' or 'map'.
+ * See the init.js.sample.
+ * And check the "plugins/{type}/example" folder for example plugin.
+ */
+
+// Wait for the page to load, then load the plugins.
+$(document).ready(function () {
+  // detect which plugins to load
+  if ($('.openwebrx-waterfall-container').length) Plugins._type = "receiver";
+  if ($('.openwebrx-map').length) Plugins._type = "map";
+  Plugins.init();
+});
+
+// Initialize the Plugins class and some defaults
+function Plugins () {}
+Plugins._initialized = false;
+Plugins._version = 0.1; // version of the plugin sub-system (keep it float)
+Plugins._enable_debug = false; // print debug to the console
+
+
+// Load plugin
+Plugins.load = async function (name) {
+  var path = 'static/plugins/' + Plugins._type + "/" + name + "/";
+  var remote = false;
+
+  // check if we are loading an url
+  if (name.toLowerCase().startsWith('https://') || name.toLowerCase().startsWith('http://')) {
+    path = name;
+    name = path.split('/').pop().split('.').slice(0, -1);
+    path = path.split('/').slice(0, -1).join('/') + '/';
+    remote = true;
+  }
+
+  // ignore plugins with same name as our methods
+  if (name === "load" || name === "init" || name === "isLoaded") {
+    console.error('PLUGINS: Cannot load plugin named "load", "isLoaded" or "init".');
+    return;
+  }
+
+  if (name in Plugins) {
+    console.warn(`PLUGINS: '${name}' is already loaded from ${Plugins[name]._script_loaded}`);
+    return;
+  }
+
+  Plugins._debug(`PLUGINS: ${remote ? '[remote]' : '[local]'} '${name}' loading.`);
+
+  var script_src = path + name + ".js";
+  var style_src = path + name + '.css';
+
+  // init plugin object with defaults
+  Plugins[name] = {
+    _version: 0,
+    _script_loaded: false,
+    _style_loaded: false,
+    _remote: remote,
+  };
+
+
+  // try to load the plugin
+  await Plugins._load_script(script_src)
+    .then(function () {
+      // plugin script loaded successfully
+      Plugins[name]._script_loaded = script_src;
+
+      // check if the plugin has init() method and execute it
+      if (typeof Plugins[name].init === 'function') {
+        if (!Plugins[name].init()) {
+          console.error(`PLUGINS: ${remote ? '[remote]' : '[local]'} '${name}' cannot initialize.`);
+          return;
+        }
+      }
+
+      // check if plugin has set the 'no_css', otherwise, load the plugin css style
+      if (!('no_css' in Plugins[name])) {
+        Plugins._load_style(style_src)
+          .then(function () {
+            Plugins[name]._style_loaded = style_src;
+            Plugins._debug(`PLUGINS: ${remote ? '[remote]' : '[local]'} '${name}' loaded.`);
```

Comment:
You can generate the "PLUGINS: ..." banner automatically, by a separate function given just plugin name.

Reply:
yes... I just added the _debug function right before the commit and I did not reworked that... will do.